ServerCore/ModelBases/EventSpecificPageModel.cs

```diff
@@ -63,5 +68,25 @@ public async Task BindModelAsync(ModelBindingContext bindingContext)
                 }
             }
         }
+
+        public class RoleBinder : IModelBinder
+        {
+            // This doesn't actually run async but the compiler complains if I try to use BindModel :(
+            public async Task BindModelAsync(ModelBindingContext bindingContext)
+            {
+                string eventRoleAsString = bindingContext.ActionContext.RouteData.Values["eventRole"] as string;
+                if (eventRoleAsString == null)
+                {
+                    eventRoleAsString = ModelBases.EventRole.play.ToString();
+                }
+                // TODO: Add auth check to make sure the user has permissions for the given eventRole
+                eventRoleAsString = eventRoleAsString.ToLower();
+
+                if (Enum.IsDefined(typeof(EventRole), eventRoleAsString))
+                {
+                    bindingContext.Result = ModelBindingResult.Success(Enum.Parse(typeof(EventRole), eventRoleAsString));
```

Comment:
Should this fail if the role isn't specified or isn't a real role?

Reply:
Will add a ToDo to fail this